python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -725,6 +733,155 @@ def slice_skip(tbl: plc.Table) -> plc.Table:
+            ):
+                _, compression_level = next(iter(compression.items()))
+                if compression_level is not None:
+                    raise NotImplementedError("compression_level is not supported.")
+        elif kind == "Json":
+            # TODO: JSON?
+            pass
+        else:
+            raise NotImplementedError(
+                f"Unhandled sink kind: {kind}"
+            )  # pragma: no cover
+
+    def get_hashable(self) -> Hashable:
+        """
+        Hashable representation of the node.
+
+        The option dictionary is serialised for hashing purposes.
+        """
+        schema_hash = tuple(self.schema.items())  # pragma: no cover
+        return (
+            type(self),
+            schema_hash,
+            self.kind,
+            self.path,
+            json.dumps(self.options),
+        )  # pragma: no cover
+
+    @classmethod
+    def do_evaluate(
+        cls,
+        schema: Schema,
+        kind: str,
+        path: str,
+        options: dict[str, Any],
+        df: DataFrame,
+    ) -> DataFrame:
+        target = plc.io.SinkInfo([path])
+
+        if options.get("mkdir", False):
+            Path(path).parent.mkdir(parents=True, exist_ok=True)
+        if kind == "Csv":
+            serialize = options["serialize_options"]
+            options = (
+                plc.io.csv.CsvWriterOptions.builder(target, df.table)
+                .include_header(options["include_header"])
+                .names(df.column_names if options["include_header"] else [])
+                .na_rep(serialize["null"])
+                .line_terminator(serialize["line_terminator"])
+                .inter_column_delimiter(chr(serialize["separator"]))
+                .build()
+            )
+            plc.io.csv.write_csv(options)
+
+        elif kind == "Parquet":
+            metadata = plc.io.types.TableInputMetadata(df.table)
+            for i, name in enumerate(df.column_names):
+                metadata.column_metadata[i].set_name(name)
+
+            builder = plc.io.parquet.ParquetWriterOptions.builder(target, df.table)
+            if (compression := options["compression"]) != "Uncompressed":
+                compression_type, _ = next(iter(compression.items()))
+                builder.compression(
+                    getattr(plc.io.types.CompressionType, compression_type.upper())
+                )
+
+            writer_options = builder.metadata(metadata).build()
+            if options["data_page_size"] is not None:
+                writer_options.set_max_page_size_bytes(options["data_page_size"])
+            if options["row_group_size"] is not None:
+                writer_options.set_row_group_size_rows(options["row_group_size"])
+
+            plc.io.parquet.write_parquet(writer_options)
+
+        elif kind == "Json":
+            metadata = plc.io.TableWithMetadata(
+                df.table, [(col, []) for col in df.column_names]
+            )
+            options = (
```

Comment:
Question: it seems like in the other cases we ask the IR for the options to use things. But here we don't. Is that because polars doesn't allow any configuration?

Reply:
Yeah I don't think polars allows for much configuration. The sink options has always been empty in the testing I've done.  I think that makes sense if you look at the [docs](https://docs.pola.rs/api/python/stable/reference/api/polars.LazyFrame.sink_ndjson.html#polars.LazyFrame.sink_ndjson). There aren't many kwargs to set compared to the other sinks